pom.xml

```diff
@@ -344,6 +343,38 @@
     </build>
 
     <profiles>
+        <profile>
+            <id>java-8</id>
+            <activation>
+                <jdk>1.8</jdk>
+            </activation>
+            <properties>
+                <maven.compiler.source>1.8</maven.compiler.source>
+                <maven.compiler.target>1.8</maven.compiler.target>
+            </properties>
+        </profile>
+        <profile>
+            <id>java-11</id>
+            <activation>
+                <jdk>[11,17)</jdk>
+            </activation>
+            <properties>
+                <maven.compiler.source>11</maven.compiler.source>
+                <maven.compiler.target>11</maven.compiler.target>
+                <maven.compiler.release>11</maven.compiler.release>
+            </properties>
+        </profile>
+        <profile>
+            <id>java-17</id>
+            <activation>
+                <jdk>[17,</jdk>
+            </activation>
+            <properties>
+                <maven.compiler.source>11</maven.compiler.source>
+                <maven.compiler.target>17</maven.compiler.target>
+                <maven.compiler.release>17</maven.compiler.release>
+            </properties>
+        </profile>
```

Comment:
Both here and in the other pom.xml ... Do we really need the profiles here? The main point of these Java 11 and Java 17 builds is to ensure that it works in the environments using Java 11 and Java 17. So I do not think we need to change the source / target versions as we want it to be as close as possible to the code you will actually release.

I do not think we did anything like this on the other Strimzi projects either.

Reply:
I found that `maven.compiler.release` option was necessary to build with Java 11 or 17. Without it I was getting an error like: `system modules path not set in conjunction with -source`

However, Java-8 build would not build if this option was present.

It could be the specifics of my local environment, but I just did whatever was necessary to get things to work.

The other thing there is the compiler source 11 in java-17 profile.  This one I kind of forgot in there.
I'll set it to 17 so it's in sync with the others.

Reply:
Not sure what `maven.compiler.release` does TBH. But maybe we can at least keep the source and target?

Reply:
I mean ideally, we would want to have a single build and then test the binaries it produces with the different JDKs. But that would be quite complicated. But I think it would be great to get at least close to it.

Reply:
In that case I would most prefer`1.8`. We could remove `java-8` profile, and remove source and target overrides from `java-11` and `java-18` profiles, but they would still need the release set to `8`.
Would that be better?